lib/content-services/document-list/components/document-list.component.ts

```diff
@@ -872,6 +868,47 @@ export class DocumentListComponent implements OnInit, OnChanges, OnDestroy, Afte
         return false;
     }
 
+    /**
+     * Updates the columns for display according to the type of nodes needed to be displayed and the custom layout defined
+     * @param nodeId The ID of the folder node to display or the reserved string alias for special sources
+     */
+    updateLayout(nodeId) {
+        if (!this.hasCustomLayout) {
+            this.setupDefaultColumns(nodeId);
+
+        } else {
+            const columns = this.columnList.columns.map(c => <DataColumn> c);
```

Comment:
I don't understand the point of this mapping.

Reply:
Because of ShareDataTableAdapter's  `setColumns` call from [here](https://github.com/Alfresco/alfresco-ng2-components/pull/2868/files/aa9fea80e94aa8c1bb9931e15fddc08c8d339e39#diff-80f7c6137233f6a2440b1e75cc0b9244R885) which requires that `columns` is of type `Array<DataColumn>`

Reply:
Hmm, correct me if I'm wrong but when it is transpiled, I think we get something like this: 
const columns = this.columnList.columns.map(c => c);
(since types are just exist in Typescript, and only important for compile time errors).

That is why I think iterating over an array (with Array.map) in runtime, just for bypassing the transpiler's type checking, is not correct.

Something like this, would it work?
const columns = <DataColumn[]>this.columnList.columns;

Reply:
I understand your concerns and that there must be a better way so as to avoid extra iterations. Although the line of code that you proposed is not working in this case, I have found [here](https://github.com/angular/angular/blob/5.1.1/packages/core/src/linker/query_list.ts#L39) that the `QueryList<DataColumnComponent>` (`columns`' type) has method `toArray` that I will use instead.